drivers/i2s/i2s_ll_stm32.c

```diff
@@ -349,9 +349,20 @@ static int i2s_stm32_trigger(const struct device *dev, enum i2s_dir dir,
 			LOG_ERR("DRAIN trigger: invalid state");
 			return -EIO;
 		}
-		stream->stream_disable(stream, dev);
-		stream->queue_drop(stream);
-		stream->state = I2S_STATE_READY;
+		if (dir == I2S_DIR_TX) {
+			if (stream->state == I2S_STATE_RUNNING) {
+				stream->state = I2S_STATE_STOPPING;
+				/*
+				 * Indicate that the transition to I2S_STATE_STOPPING
+				 * is triggered by DRAIN command
+				 */
+				stream->sample_tx_stop_drain = true;
+			} else {
+				stream->stream_disable(stream, dev);
+				stream->queue_drop(stream);
+				stream->state = I2S_STATE_READY;
+			}
```

Comment:
Non-RUNNING case already treated just above:
```
		if (stream->state != I2S_STATE_RUNNING) {
			irq_unlock(key);
			LOG_ERR("DRAIN trigger: invalid state");
			return -EIO;
		}
```

Reply:
+1

The check here should probably be something like `if (/* stream queue not empty || stream DMA busy */)` instead.